index.js

```diff
@@ -290,7 +290,12 @@ function parse (args, opts) {
   // e.g., --foo apple banana cat becomes ["apple", "banana", "cat"]
   function eatArray (i, key, args) {
     for (var ii = i + 1; ii < args.length; ii++) {
-      if (/^-/.test(args[ii])) break
+      if (/^-/.test(args[ii])) {
+        if (args[ii - 1].replace('-', '') === key) {
```

Comment:
whoops, you're right @maxrimue; I think we should add an extra test for this edge-case that I would have broke.

Could we instead do this:

```js
  // if an option is an array, eat all non-hyphenated arguments
  // following it... YUM!
  // e.g., --foo apple banana cat becomes ["apple", "banana", "cat"]
  function eatArray (i, key, args) {
    var start  = i + 1
    for (var ii = i + 1; ii < args.length; ii++) {
      if (/^-/.test(args[ii])) {
        if (ii === start) setArg(key, defaultForType('array'))
        break
      }
      i = ii
      setArg(key, args[ii])
    }

    return i
  }
```

mainly I think it reads clearer, you want to return `[]` if `0` positional arguments follow the starting key?